common/internal_model/src/main/scala/uk/ac/wellcome/models/work/internal/Location2.scala

```diff
@@ -0,0 +1,37 @@
+package uk.ac.wellcome.models.work.internal
+
+sealed trait Location2 {
+  val accessConditions: List[AccessCondition]
+}
+object Location2 {
+  case class OpenShelves(accessConditions: List[AccessCondition],
+                         shelfmark: String,
+                         sectionName: String)
+      extends Location2
+  case class ClosedStores(accessConditions: List[AccessCondition])
+      extends Location2
```

Comment:
Regarding naming, in my opinion the singular makes sense for these: (i.e.`OpenShelf` and `ClosedStore`): the location is an open shelf location, a closed store location or a digital resource location.

Reply:
The way we speak about it is definitely in the plural as that is the name of the location.
we would not say give us books that are on an open shelf, but rather give us books that are on the open shelves.

I think if singularise the word we make the location too specific.

Reply:
Yeh I guess I am thinking it more from a semantic angle than strictly to do with what language is used. In general we don't really use pluralised class names, and find it strange having an ADT defined like that. Regarding the language, these two statements don't really  conflict with each other in my opinion:

* Give us the books that are on the open shelves
* Give us the books that have the open shelf location type

Reply:
Semantically they are the same, but we never speak about an Open shelf.

I would rather keep language the same - these locations are known and have a way for them to be described and I would prefer to keep that. The name of the location is the Open shelves. An open shelf isn't used in the current vernacular.

I can't think that having a rule to never pluralise is useful or makes sense in helping with understanding or reading code.

As another example if we were talking about the location "The marshes", we wouldn't singularise that to "The Marsh".

Reply:
OK I get your marshes example, it still seems a bit weird to me but acknowledge that is probably to do with me not really being familiar with the terminology :+1:

Reply:
Sorry to be contrived with the example - but the point I was trying to get across is the location is a know entity and is called the `Open shelves` which is what I am trying to encapsulate in this type class. Maybe because it's not a abstract concept but rather a representation of a known entity.